tests/playwright/src/model/pages/deploy-to-kubernetes-page.ts

```diff
@@ -0,0 +1,119 @@
+    this.deployButton = this.content.getByRole('button', { name: 'Deploy' });
+    this.doneButton = this.content.getByRole('button', { name: 'Done' });
+    this.servicesCheckbox = this.content.getByRole('checkbox', { name: 'Use Services' });
+    this.restrictedContextCheckbox = this.content.getByRole('checkbox', { name: 'Use restricted security context' });
+    this.createIngressCheckbox = this.content.getByRole('checkbox', { name: 'Create Ingress' });
+    this.selectPortCombobox = this.content.getByRole('combobox', { name: 'Select a Port' });
+  }
+
+  public async deployPod(
+    name: string,
+    {
+      useKubernetesServices,
+      useRestrictedSecurityContext,
+      useKubernetesIngress,
+      containerExposedPort,
+    }: DeployPodOptions = {},
+    context: string,
+    namespace: string = 'default',
+  ): Promise<void> {
+    await playExpect(this.podName).toBeVisible();
+    await this.podName.fill(name);
+
+    await playExpect(this.servicesCheckbox).toBeEnabled();
+    if (useKubernetesServices) {
+      await this.servicesCheckbox.check();
+      await playExpect(this.servicesCheckbox).toBeChecked();
+    } else {
+      await this.servicesCheckbox.uncheck();
+      await playExpect(this.servicesCheckbox).not.toBeChecked();
+    }
+
+    await playExpect(this.restrictedContextCheckbox).toBeEnabled();
+    if (useRestrictedSecurityContext) {
+      await this.restrictedContextCheckbox.check();
+      await playExpect(this.restrictedContextCheckbox).toBeChecked();
+    } else {
+      await this.restrictedContextCheckbox.uncheck();
+      await playExpect(this.restrictedContextCheckbox).not.toBeChecked();
+    }
+
+    await playExpect(this.createIngressCheckbox).toBeEnabled();
+    if (useKubernetesIngress) {
+      await this.createIngressCheckbox.check();
+      await playExpect(this.createIngressCheckbox).toBeChecked();
+      if (containerExposedPort) {
+        await this.selectExposedPort(containerExposedPort);
+      }
+    } else {
+      await this.createIngressCheckbox.uncheck();
+      await playExpect(this.createIngressCheckbox).not.toBeChecked();
+    }
+
+    await playExpect(this.kubernetesContext).toHaveValue(context);
+    await playExpect(this.namespaceCombobox).toBeVisible();
+    const currentNamespace = await this.namespaceCombobox.inputValue();
+    if (currentNamespace !== namespace) {
+      const namespaceOptions = await this.namespaceCombobox.locator('option').allInnerTexts();
+      if (!namespaceOptions.includes(namespace)) {
+        throw new Error(`${namespace} doesn't exist`);
+      }
+      await this.namespaceCombobox.selectOption({ value: namespace });
+      await playExpect(this.namespaceCombobox).toHaveValue(namespace);
+    }
+
+    await playExpect(this.deployButton).toBeEnabled();
+    await this.deployButton.click();
+    await playExpect(this.doneButton).toBeVisible({ timeout: 30000 });
```

Comment:
please make it a parameter with default value. I think that 30 sec. wont be enough.